### PR TITLE
[WIP] qt5: update to 5.12.1 

### PR DESCRIPTION
**Pull request by Johnnynator**
Status:

| Arch | Build | Runs |
| --- | --- | --- |
| x86_64 | x | x |
| x86_64-musl | x | webengine broken |
| i686 | x  |  |
| aarch64 |  |  |
| aarch64-musl |  |  |
| armv7l  |  |  |
| armv7l-musl  |  |  |
| armv6l |  |  |
| armv6l-musl |  |  |
| armv5 |  |  |
| ppc64 |  |  |

Test Repo: https://void.johnnynator.dev/qt5.12/

### CODE REVIEW

File: srcpkgs/qt5/template [hunk 1]
```diff
@@ -425,7 +428,6 @@ do_install() {
 	cd ${wrksrc}/qtbase
 	make INSTALL_ROOT=${DESTDIR} install
 
-	vlicense ${wrksrc}/LGPL_EXCEPTION.txt
```
Comment (pullmoll) [inline srcpkgs/qt5/template]: Shouldn't we include the LGPL_EXPECTION.txt or was it removed?

Comment (Johnnynator) [reply srcpkgs/qt5/template]: File was removed

File: srcpkgs/qt5/template [hunk 2]
```diff
@@ -287,6 +289,7 @@ do_configure() {
 		done
 
 		if [ "$build_option_webengine" ]; then
+			:
```
[the remark this answers is not in the document]
Comment (Johnnynator) [reply srcpkgs/qt5/template]: Leftover.

File: srcpkgs/qt5/template (hunk 1 above)
Comment (Johnnynator) [reply srcpkgs/qt5/template]: Readded it from one of the sub dir

Comment (pullmoll) [reply srcpkgs/qt5/template]: Well, I tried to build your PR and also realized it didn't work. I guess we need to find out why the file was removed.

Comment (pullmoll) [reply srcpkgs/qt5/template]: Ok, here's the commit removing it:
https://code.qt.io/cgit/qt/qt5.git/commit/?h=dev&id=21a37fe009d28e394d154c4069de332075814f8f

This means we need to update `license=` to use `LGPL-3.0-or-later`

File: srcpkgs/qt5/template [hunk 3]
```diff
@@ -21,7 +21,7 @@ maintainer="Jürgen Buchmüller <pullmoll@t-online.de>"
 license="GPL-3.0-or-later, LGPL-2.1-or-later"
```
Comment (pullmoll) [inline srcpkgs/qt5/template]: `LGPL-3.0-or-later`

**Comment by pullmoll** [comment]
Required addition to `common/shlibs`:
```
WARNING: qt5-declarative-5.12.1_1: libQt5QuickShapes.so.5 not found in common/shlibs
```

File: srcpkgs/qt5/template (hunk 1 above)
Comment (Johnnynator) [reply srcpkgs/qt5/template]: > Well, I tried to build your PR and also realized it didn't work.

It should work fine for at least x86_64 glibc. Haven't gotten cross builds figured out yet. I work currently on getting webengine patched for musl

**Comment by q66** [comment]
While building something else, I found that with 5.12.2, the `cups` feature is enabled in `QtPrintSupport` which means `qprint_p.h` now includes a header from `cups-devel` without this being a dependency, what do we do?

**Comment by q66** [comment]
Also, `qt5-webkit` will need bumping as I found that it stores the version number of qt5 (the major and minor version, but not patch) in its cmake scripts, which results in e.g. `kdewebkit` failing to configure with unbumped `qt5-webkit`.

**Comment by Johnnynator** [comment]
I got carries away and had no time to finish this. I now have a at 5.13 branch that has more stuff fixed. Will open a new PR later™.

**Pull request closed without merging**